### PR TITLE
VACMS-5844: Install and configure Admin feedback module.

### LINKED ISSUE #5811: Finish designing and Implement a "Was this helpful?" widget
As a KB manager, i know if my content is helping users. 

<details><summary>Designs so far, screenshot</summary>

Widget ideas

Linking_from_the_editorial_experience_–_Figma

Feedback post widget ideas

Linking_from_the_editorial_experience_–_Figma

</details>

[Link to figma](https://www.figma.com/file/PySxm5aQeWCQvYqZpdkYJj/Linking-from-the-editorial-experience?node-id=0%3A1)

^^^ needs edits. Plz check w/ Cindy & Kauff first before starting!

## ACs
- [x] Find best drupal contrib architecture for this
- [x] evaluate workflow and UI with design team
- [x] We can collect metrics about each node, and available to knowledge base admins
- [x] Additinoal feedback comment box is a nice to have.

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Closes #5811

## Testing done
manual, behat

## Screenshots
Front end:
image

Dashboard: 
image

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As a cms user
1. Navigate to a knowledgebase page (`/help/cms-basics/how-to-log-inout`)
   - [x] Validate that the 'was this helpful' block appears at the bottom of the page
   - [x] Click a button and validate a thank you message appears
   - [x] If 'no' clicked, validate a message appears to go to help desk (this content is not 'final' yet)
2. Navigate to the dashboard (`/admin/content/feedback`)
   - [x] Validate the dashboard is available and has responses
   - [ ] View matches as many of our Views standards as possible, given the odd nature of this non-content type View.
3. Then validate Acceptance Criteria from issue
   - [x] We can collect metrics about each node, and available to knowledge base admins
   - [x] Add helpdesk link for 'no' responses 
   - [x] Style yes/no as secondary icon buttons 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -48,6 +49,7 @@
         "drupal/ctools_block": "^3.2",
         "drupal/danse": "^2.1",
         "drupal/danse_content_moderation": "^2.0",
+        "drupal/did_this_help": "^2.0",
```
Comment (JayDarnell) [inline composer.json:52]: Was this accidentally left in? It looks like you may have been exploring both modules but only ended up needing one of them.

File: docroot/themes/custom/vagovclaro/vagovclaro.theme [hunk 2]
```diff
@@ -359,8 +359,11 @@ function vagovclaro_preprocess_page__documentation_page(&$variables) {
   }
 
   // Don't show header search on KB homepage, which has form visible lower.
-  $node = $route_match->getParameter('node');
-  $nid = $node->id();
+  // Changing to be a string so the feedback tab works
+  // (will always return a string or null, instead of mixed or null)
+  // $node = $route_match->getParameter('node');
```
Comment (JayDarnell) [inline docroot/themes/custom/vagovclaro/vagovclaro.theme:364]: Lines 364 and 365 look like they could be deleted altogether.

**Review by JayDarnell — CHANGES_REQUESTED** [review]
Just a couple items that stood out to me. The rest looks good as far as I can tell. I haven't QAed this in a test environment to verify it works. I've only looked at the code thus far.

File: composer.json (hunk 1 above)
Comment (msbtterswrth) [reply composer.json:52]: great catch!

File: docroot/themes/custom/vagovclaro/vagovclaro.theme [hunk 3]
```diff
@@ -359,8 +359,9 @@ function vagovclaro_preprocess_page__documentation_page(&$variables) {
   }
 
   // Don't show header search on KB homepage, which has form visible lower.
-  $node = $route_match->getParameter('node');
-  $nid = $node->id();
+  // Changing to be a string so the feedback tab works
+  // (will always return a string or null, instead of mixed or null)
```
[the remark this answers is not in the document]
Comment (msbtterswrth) [reply docroot/themes/custom/vagovclaro/vagovclaro.theme:363]: I had also left in the code i changed (commented out), but Jay suggested I remove the old code. This comment refers to the change from `->getParameter` to `->getRawParameter`. Wanted to leave a note for Tanner so he knew what was going on there. Happy to take this comment out too, if you suggest.

File: composer.lock [hunk 4]
```diff
@@ -4835,6 +4887,50 @@
...
+                    "version": "2.0.3",
+                    "datestamp": "1615281905",
+                    "security-coverage": {
+                        "status": "not-covered",
+                        "message": "Project has not opted into security advisory coverage!"
+                    }
+                }
+            },
+            "notification-url": "https://packages.drupal.org/8/downloads",
+            "license": [
```
Comment (swirtSJW) [inline composer.lock:4919]: @msbtterswrth  Looks like you removed this from composer.json  by hand from Jay's feedback but did not run `composer remove drupal/did_this_help`
For patches you can tweak the .json by hand, but for additions removals or update you have to use composer commands to make it happen.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Overall looks good.  Just one issue that needs addressing.

File: composer.lock (hunk 4 above)
Comment (msbtterswrth) [reply composer.lock:4919]: oops!

**Comment by msbtterswrth** [comment]
@swirtSJW hold on on merging this until we get the final copy from @rachel-kauff, it should be today or tomorrow, but likely past cutoff for deployment today.

**Comment by rachel-kauff** [comment]
@msbtterswrth I noticed the styling looks funny for the 'submit' button for the "no / add more feedback" step.

**Comment by msbtterswrth** [comment]
> @msbtterswrth I noticed the styling & alignment looks funny for the 'submit' button for the "no / add more feedback" step. Screen Shot 2022-04-26 at 12 53 15 PM

Ah good catch @rachel-kauff, I didn't even look at those, bc we weren't using that feature but I can fix that up real quick. Same blue outline button?

**Comment by rachel-kauff** [comment]
> Ah good catch @rachel-kauff, I didn't even look at those, bc we weren't using that feature but I can fix that up real quick. Same blue outline button?

Yes! Secondary button pattern, same size and all.

**Comment by rachel-kauff** [comment]
@msbtterswrth You mentioned that it might be possible to link to the help desk. Does that need to get hard coded in?

**Comment by rachel-kauff** [comment]
@msbtterswrth Other than the link, the microcopy is ready to go! Happy to iterate based on how we need to do the link.

**Comment by msbtterswrth** [comment]
Ok @swirtSJW this is good to go!

**Review by swirtSJW — COMMENTED** [review]
Code looks good.
Getting hung up on View standards.  This view does not seem to adhere to our standards
image


The export seems to not be valid csv
image
This is a semicolon-sv

**Comment by swirtSJW** [comment]
Looks like one of the fixes didn't stick
image

**Comment by swirtSJW** [comment]
And this csv is still a  semi-colon-SV
image

**Comment by swirtSJW** [comment]
Interesting, it looks like that export button is hardcoded and not done with an attached view.   Fixing that to be a proper CSV can be ignored as it is not in our immediate  control.

**Comment by swirtSJW** [comment]
I created an [upstream issue for the download](https://www.drupal.org/project/admin_feedback/issues/3277800) but I do not feel like this is anything we need to fix at this time.  We can get around it if needed just by  altering the csv import settings on our spreadsheet or by doing a find `";"` and replace `"," on the file in a text editor.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
I think we are still missing the View standards
image
The edit should come after the title and be a button like all the rest of our admin/content/* Views
image

**Comment by msbtterswrth** [comment]
My apologies @swirtSJW, I very misunderstood our earlier convo. We're all good to go now:

image

**Review by swirtSJW — APPROVED** [review]
Tests out good.  Thanks @msbtterswrth  that is a great addition to the site.
image

**Comment by swirtSJW** [comment]
Test failure is an unrelated deadlock issue
image


image

@ndouglas ^^ Just for your awareness for tracking failures that might be due to parallelization ??

**Review by swirtSJW — CHANGES_REQUESTED** [review]
This is on hold.   While we investigate that this logs an alert for every interaction with the View
image

File: config/sync/views.view.feedback.yml [hunk 5]
```diff
@@ -0,0 +1,1541 @@
...
+          table: views
+          field: area
+          relationship: none
+          group_type: group
+          admin_label: ''
+          plugin_id: text
+          empty: false
+          content:
+            value: "There is one line <strong>per translation</strong> below and the overall score is for each translation!\r\nSo, a node's feedback can be different between every language."
+            format: basic_html
```
Comment (swirtSJW) [inline config/sync/views.view.feedback.yml:960]: I think this is the problem.  We don't have a basic_html as a format.  We need to set this to one of our existing formats

Comment (swirtSJW) [inline config/sync/views.view.feedback.yml:960]:
```suggestion
            format: rich_text_limited
```

**Review by swirtSJW — APPROVED** [review]
Changing the text format resolved the dblog alerts.  This is ready to go

**Pull request merged**